### PR TITLE
Add isSupported to stageView

### PR DESCRIPTION
**Pull request by richardle798**

### CODE REVIEW

File: packages/teams-js/src/public/stageView.ts [hunk 1]
```diff
@@ -53,11 +54,24 @@ export namespace stageView {
...
       if (!stageViewParams) {
         throw new Error('[stageView.open] Stage view params cannot be null');
       }
 
       resolve(sendAndHandleSdkError('stageView.open', stageViewParams));
     });
   }
+
+  /**
+   * Checks if stageView capability is supported by the host
```
Comment (erinha) [inline packages/teams-js/src/public/stageView.ts:70]: While you're here, can you add @beta tags to this doc comment and to the top of the namespace please? (I assume that this capability is still under development. If not, then we should not add @beta and instead should unhide the doc comments for `open`.)

File: change/@microsoft-teams-js-891e4235-67b9-4b5a-b0f7-d503a900afca.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "minor",
+  "comment": "Add isSupported to stageView",
```
Comment (erinha) [inline change/@microsoft-teams-js-891e4235-67b9-4b5a-b0f7-d503a900afca.json:3]: Please update the comment to the following: "Added `isSupported` to `stageView`" -- please enclose "isSupported" and "stageView" in backticks and use past tense. (These guidelines are also documented in the CONTRIBUTING.md file in the monorepo root.) Thanks!

**Review by erinha — CHANGES_REQUESTED** [review]
Two small comments about change file and doc comments.

File: packages/teams-js/src/public/stageView.ts (hunk 1 above)
Comment (richardle798) [reply packages/teams-js/src/public/stageView.ts:72]: Added beta tag to the stageView namespace to follow convention of other APIs in beta

**Review by erinha — APPROVED** [review]
🚀

File: packages/teams-js/src/public/stageView.ts [hunk 3]
```diff
@@ -53,11 +54,24 @@ export namespace stageView {
     return new Promise(resolve => {
       ensureInitialized(FrameContexts.content);
 
+      if (!isSupported()) {
```
[the remark this answers is not in the document]
Comment (richardle798) [reply packages/teams-js/src/public/stageView.ts:60]: Hmm in principle, I wouldn't want to rely on my consumer needing call the API beforehand.  I also, checked in other areas of the codebase, and it seems to be pretty commonly checked by other APIs: https://github.com/OfficeDev/microsoft-teams-library-js/search?q=isSupported

**Review by ht4963-ms — APPROVED** [review]

File: packages/teams-js/src/public/stageView.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (ht4963-ms) [reply packages/teams-js/src/public/stageView.ts:60]: I'm seeing this in about half of the apis. It seems we should have a standardized way of handling this.

[the remark this answers is not in the document]
Comment (erinha) [reply packages/teams-js/src/public/stageView.ts:60]: The pattern we are moving forward with is to explicitly check in the teams-js SDK for isSupported and throw errorNotSupportedOnPlatform if it is false. This is to ensure that we fail fast before making a network call and have a consistent experience with respect to supported capabilities in the host. The app is encouraged to also check for `isSupported` before calling functions so they can appropriately show or hide UI, etc.

[the remark this answers is not in the document]
Comment (erinha) [reply packages/teams-js/src/public/stageView.ts:60]: @ht4963-ms I would love to see the places it's missing. We did a push before releasing 2.0.0 to add `isSupported` before each of the APIs for the capabilities that have an `isSupported` function to call. There are some namespaces that haven't been converted to capabilities (e.g., media, files, meeting) so those functions won't call `isSupported` first, But if there is an `isSupported` function in the namespace and functions also in the namespace aren't calling it, that is a bug.

File: apps/teams-test-app/src/components/StageViewAPIs.tsx [hunk 4]
```diff
@@ -1,7 +1,14 @@
...
 import React, { ReactElement } from 'react';
 
-import { ApiWithTextInput } from './utils';
+import { ApiWithoutInput, ApiWithTextInput } from './utils';
+
+const CheckStageViewCapability = (): React.ReactElement =>
+  ApiWithoutInput({
+    name: 'checkStageViewCapability',
+    title: 'Check StageView Capability',
+    onClick: async () => `StageView ${stageView.isSupported() ? 'is' : 'is not'} supported`,
```
[the remark this answers is not in the document]
Comment (richardle798) [reply apps/teams-test-app/src/components/StageViewAPIs.tsx:10]: All other check isSupported onClicks are async

**Pull request merged**